packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -1542,6 +1543,14 @@ export interface BuildEnvironment {
    */
   readonly privileged?: boolean;
 
+  /**
+   * The ARN of the Amazon S3 bucket, path prefix, and object key that contains the
+   * PEM-encoded certificate for the build project
+   *
+   * @default - No external certificate is added to the project
+   */
+  readonly certificate?: string;
```

Comment:
I'm wondering whether a string is the best way to model this...? Seems a little looks for my taste.

I see 2 possible alternatives:

1. Introduce a dedicated interface for this property, something like `BuildCertificate`, that takes in an `s3.IBucket`, a string for the `key`, and maybe an optional `version`?
2. Use the existing [`Location` interface](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-s3.Location.html) from the S3 module. There's a small downside that it takes in a `bucketName` as just a string, but maybe that's fine - we can show an example in the ReadMe of importing a Bucket by ARN, and then using it to pass the `bucketName` property as `bucket.bucketName`.

I'll let you decide which option you like more!

Reply:
Ah thanks! I was looking for something better but couldn't find existing classes that met the need and wasn't sure about adding another interface. I like option 2 better since it already exists.

However, arns do not support versions and the BuildEnvironment certificate at the cloudformation level only accepts a string value that is an arn. It seems like there's no way to support a version at the moment for the certificate. Because of this I'll create a new interface that takes in an IBucket and key only so people don't think they can use a version when it's not supported.

Let me know if I'm wrong about the version? If so I'll use the existing interface. Won't be hard to do either way.

Reply:
Yeah, let's go with a new interface here.